### PR TITLE
fix: smoothen battery level before showing it to user

### PR DESCRIPTION
**Pull request by thiswillbeyourgithub**
Signed-off-by: thiswillbeyourgithub <github@32mail.33mail.comm>

edit : I reworked the code to smoothen the voltage instead of the level

### CODE REVIEW

File: wasp/drivers/battery.py [hunk 1]
```diff
@@ -50,26 +51,34 @@ def voltage_mv(self):
...
         """
         raw = self._battery.read_u16()
-        return (2 * 3300 * raw) // 65535
+        mv = (2 * 3300 * raw) // 65535
+
+        if mv not in self._cache:
+            self._cache.add(mv)
+            while len(self._cache) > 2:
+                self._cache.remove(max(self._cache))
+        return min(self._cache)
```
Comment (fgaz) [inline wasp/drivers/battery.py:68]: won't this always keep the minimum voltage?
for example
|loop|mv|cache|min|
|-|-|-|-|
|0|1|{1}|1|
|1|2|{1,2}|1|
|2|5|{1,2}|1|
|...|...|{1,2}|1|

**Comment by thiswillbeyourgithub** [comment]
I think you're right. Indeed in some cases the battery level is stuck at the lowest possible level until I reboot but I thought it was related to another change. I'm now trying with using `.pop()` to remove the oldest value and returning the mean of the `_cache` and that should be better. Should I open a new PR for this?

**Comment by fgaz** [comment]
Does the voltage reading really fluctuate so much that it needs smoothing? In my experience it's only noticeable if you look at the raw mv value.

**Comment by thiswillbeyourgithub** [comment]
Personally I use the mV value and the smoothing is necessary.

**Comment by thiswillbeyourgithub** [comment]
Shoot I just realized that .pop() in sets is not implementd the same way in python as in micropython. Apparently this always pops the last value instead of the oldest. I reverted to using an array in my own fork and will probably do a PR in the future

**Comment by thiswillbeyourgithub** [comment]
Sorry everyone!

I noticed this some time ago so I PR'd in #387  my current version of the drivers/battery file.

**Pull request merged**